SeaBattle.API/Controllers/EngineController.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using EngineNamespace;
+using BoardNamespace;
+using SeaBattle.Engine.Interfaces;
+using SeaBattle.Values;
+using SeaBattle.API.Infrastructure;
+using Microsoft.AspNetCore.Cors;
+using SeaBattle.API.Services;
+using SeaBattle.API.Domains.Engine.Models;
+using SeaBattle.API.Controllers;
+
+namespace WebApp.Controllers
+{
+    [ApiController]
+    [Route("api/seabattle")]
+    [EnableCors("AllowSpecificOrigin")]
+    public class GameController : ControllerBase
+    {
+        private readonly EngineService _engineService;
+        private readonly ILogger<GameController> _logger;
+
+        public GameController(ILogger<GameController> logger, EngineService engine)
+        {
+            _engineService = engine;
+            _logger = logger;
+            
+        }
+
+        [HttpPost("shoot")]
+        public async Task<IActionResult> Shoot([FromBody] CoordinateDTO coords)
+        {
+            var inputHandler = new WebInput(new Coordinates(coords.x, coords.y));
+            _engineService.engine.SetInputHandler(inputHandler); // Добавьте метод SetInputHandler в Engine
```

Comment:
Issue: You don't need to create a service every time. Consider to have a WebInput service, which is created only once and then resolved by Dependency Injection when needed (like you did for the `EngineService`)

Reply:
To do that you will need to redesign service or change its signature

Reply:
you mean builder.Services.AddScoped<>()?